lib/PoolClusterFarm.js

```diff
@@ -0,0 +1,120 @@
+var Pool          = require('./Pool');
+var PoolConfig    = require('./PoolConfig');
+var PoolNamespace = require('./PoolNamespace');
+var PoolSelector  = require('./PoolSelector');
+var Util          = require('util');
+var EventEmitter  = require('events').EventEmitter;
+
+/**
+ * PoolClusterFarm
+ * @constructor
+ * @param {object} [config] The pool cluster farm configuration
+ * @public
+ */
+function PoolClusterFarm(config) {
+  EventEmitter.call(this);
+
+  config = config || {};
+  this._servers = [];
+  this._numMasters = 0;
+  this._numSlaves = 0;
+  this._last_index = 0;
+  this.debug = config.debug || false
+}
+
+Util.inherits(PoolClusterFarm,EventEmitter);
+
+PoolClusterFarm.prototype.query = function query(sql,values,cb) {
+	var self = this
+
+	if (!this._numMasters || this._numMasters < 1) {
+		if (cb && typeof cb == 'function') { return cb('No servers assigned to PoolClusterFarm. Please verify...'); }
+		else { return values('No servers assigned to PoolClusterFarm. Please verify...'); }
+	}
+
+	if (typeof sql != 'string' || sql.length < 7) {
+		if (cb && typeof cb == 'function') { return cb('Invalid sql statement. Please verify...'); }
+		else { return values('Invalid sql statement. Please verify...'); }
+	}
+
+	if (sql.substr(0,6).toLowerCase() == 'select') {
```

Comment:
@dougwilson hey man the only way I can think to do this is if someone is using mostly stored procedures then they're just going to have to run all masters. It will still be pretty fast. I can't think of a way for me to know what kind of query/queries are in the stored procedure. Do you know of any way to describe a stored procedure and then do logic based on the result without significantly slowing down the module? thanks!